### PR TITLE
 UCP: fix err handling on wireup

### PR DESCRIPTION
**Pull request by evgeny-leksikov**
## What
Fixes https://github.com/openucx/ucx/issues/2981 by
 - fixing err handling on wireup
 - fixing test_ucp_sockaddr_with_rma_atomic.wireup

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8083/ for details (Mellanox internal link).

File: src/ucp/wireup/wireup.c [hunk 1]
```diff
@@ -738,7 +736,19 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
     status = ucp_wireup_select_lanes(ep, params, ep_init_flags, address_count,
                                      address_list, addr_indices, &key);
     if (status != UCS_OK) {
-        goto err;
+        if (ep->flags & UCP_EP_FLAG_USED) {
+            failed_lane = 0; /* lanes aren't selected but need to invoke err
```
Comment (yosefe) [inline src/ucp/wireup/wireup.c]: why lane number doesn't matter, you are assuming it exists and set it to failed?
maybe ucp_worker_set_ep_failed should handle failed_lane=NULL to treat all lanes as failed?

File: src/ucp/wireup/wireup.c [hunk 2]
```diff
@@ -738,7 +736,19 @@ ucs_status_t ucp_wireup_init_lanes(ucp_ep_h ep, const ucp_ep_params_t *params,
...
+                                handler, lane number doesn't matter here */
+            goto err;
+        }
+
+        for (lane = 0; lane < ucp_ep_num_lanes(ep); ++lane) {
```
Comment (yosefe) [inline src/ucp/wireup/wireup.c]: why don't we call ucp_worker_set_ep_failed() if ep is not used?
maybe ucp_worker_set_ep_failed() could handle this case as well?

File: test/gtest/ucp/test_ucp_sockaddr.cc [hunk 3]
```diff
@@ -279,11 +279,11 @@ class test_ucp_sockaddr : public ucp_test {
     {
         ucs_time_t time_limit = ucs_get_time() + ucs_time_from_sec(UCP_TEST_TIMEOUT_IN_SEC);
 
-        while ((receiver().get_num_eps() == 0) && !is_failed() &&
+        while ((receiver().get_num_eps() == 0) && (err_handler_count == 0) &&
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_sockaddr.cc]: what is reason for this change?

Comment (evgeny-leksikov) [reply test/gtest/ucp/test_ucp_sockaddr.cc]: When server can't create endpoint to the client, the client gets rejected error code. is_failed returns true for unreachable err code

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8100/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8127/ for details (Mellanox internal link).

File: test/gtest/ucp/test_ucp_sockaddr.cc (hunk 3 above)
Comment (yosefe) [reply test/gtest/ucp/test_ucp_sockaddr.cc]: maybe need to change the naming here? (e.g is_failed() to check all errs, is_unreachable() to check unreachable err code)?

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/8137/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:bgate:retest

**Comment by yosefe** [comment]
bgate test node is down

**Pull request merged**